examples/Training/java/src/training/ROIFolders.java

```diff
@@ -0,0 +1,185 @@
+import omero.gateway.LoginCredentials;
+import omero.gateway.SecurityContext;
+import omero.gateway.facility.BrowseFacility;
+import omero.gateway.facility.ROIFacility;
+import omero.gateway.model.ExperimenterData;
+import omero.gateway.model.FolderData;
+import omero.gateway.model.ImageData;
+import omero.gateway.model.ROIData;
+import omero.gateway.model.ROIResult;
+import omero.gateway.model.RectangleData;
+import omero.log.SimpleLogger;
+
+/**
+ * Sample code showing how to organize ROIs in Folders
+ *
+ * @author Dominik Lindner &nbsp;&nbsp;&nbsp;&nbsp; <a
+ *         href="mailto:d.lindner@dundee.ac.uk">d.lindner@dundee.ac.uk</a>
+ */
+public class ROIFolders {
+
+    // The value used if the configuration file is not used. To edit*/
+    /** The server address. */
+    private static String hostName = "serverName";
+
+    /** The username. */
+    private static String userName = "userName";
+
+    /** The password. */
+    private static String password = "password";
+
+    /** Information to edit. */
+    private static long imageId = 1;
+    // end edit
+
+    private ImageData image;
+
+    private Gateway gateway;
+
+    private SecurityContext ctx;
+
+    /**
+     * start-code
+     */
+
+    /**
+     * Loads the image.
+     * 
+     * @param imageID
+     *            The id of the image to load.
+     * @return See above.
+     */
+    private ImageData loadImage(long imageID) throws Exception {
+        BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
+        return browse.getImage(ctx, imageID);
+    }
+
+    /** Creates rois and organize in folders */
+    private void createROIs() throws Exception {
+        ROIFacility roifac = gateway.getFacility(ROIFacility.class);
+
+        // Create 3 Rectangle ROIs
+
+        Collection<ROIData> rois = new ArrayList<ROIData>();
+
+        ROIData roi = new ROIData();
+        roi.setImage(image.asImage());
+        RectangleData rectangleData = new RectangleData(10, 10, 10, 10);
+        rectangleData.setZ(0);
+        rectangleData.setT(0);
+        roi.addShapeData(rectangleData);
+        rois.add(roi);
+
+        roi = new ROIData();
+        roi.setImage(image.asImage());
+        rectangleData = new RectangleData(20, 20, 10, 10);
+        rectangleData.setZ(0);
+        rectangleData.setT(0);
+        roi.addShapeData(rectangleData);
+        rois.add(roi);
+
+        roi = new ROIData();
+        roi.setImage(image.asImage());
+        rectangleData = new RectangleData(30, 30, 10, 10);
+        rectangleData.setZ(0);
+        rectangleData.setT(0);
+        roi.addShapeData(rectangleData);
+        rois.add(roi);
+
+        rois = roifac.saveROIs(ctx, image.getId(), rois);
+
+        // Add each ROI to a different folder
+        for (ROIData r : rois) {
+            FolderData folder = new FolderData();
+            folder.setName("Folder for ROI " + r.getId());
+            roifac.addRoisToFolders(ctx, imageId, Arrays.asList(r),
```

Comment:
you need to use ``image.getID()`` instead of ``imageId`` otherwise this will throw
``
Cannot Save the ROI for image: 1omero.ValidationException
    serverStackTrace = "ome.conditions.ValidationException: No row with the given identifier exists: [ome.model.core.Image#1]; nested exception is org.hibernate.ObjectNotFoundException: No row with the given identifier exists: [ome.model.core.Image#1]

``